### PR TITLE
Zebra - F# / Aspnetcore web framework 

### PR DESCRIPTION
**Pull request by gerardtoconnor**
This is my new web framework based on the following blog post:

https://medium.com/@gerardtoconnor/racing-the-zebra-benchmark-performance-architecture-for-f-web-server-58dd922f5cfe

We would like to get more F# frameworks in the mix and Zebra *should* hopefully perform a little better thanks to reduced captures & allocations as well as a shared async state-machine.

@benaadams @forki @NinoFloris

### CODE REVIEW

**Comment by forki** [comment]
where can we see results?

**Comment by gerardtoconnor** [comment]
Seem to be run every three days here https://tfb-status.techempower.com/
If the PR got accepted today, means it would make tomorrows run

**Comment by gerardtoconnor** [comment]
FSharp passed CI testing, some other (untouched) frameworks are failing but they seem to have been failing in the recent run a few days ago also.

**Comment by gerardtoconnor** [comment]
Hmmm ... while we wait, @forki @NinoFloris any feedback, any glaring issues other then some of the code being painfully explicit etc, am I missing any critical functionality?

File: frameworks/CSharp/aspnet-mono-ngx/src/obj/Debug/Benchmarks.csproj.CoreCompileInputs.cache [hunk 1]
```diff
@@ -0,0 +1 @@
+0418627bce649e969f6668eb93deacedd9d17f8a
```
Comment (michaelhixson) [inline frameworks/CSharp/aspnet-mono-ngx/src/obj/Debug/Benchmarks.csproj.CoreCompileInputs.cache]: Could you undo the changes to files in `frameworks/CSharp/aspnet-mono-ngx`, assuming those changes are unintentional?

File: frameworks/FSharp/Zebra/src/App/Parsers.fs [hunk 2]
```diff
@@ -0,0 +1,162 @@
...
+
+/// Private Range Parsers that quickly try parse over matched range (all r.Finish checked before running in preceeding functions)
+
+let stringParse (path : string,r:Range) = path.Substring(r.Start, r.Finish - r.Start + 1) |> rtrn
+
+let charParse (path : string,r:Range) = path.[r.Start] |> rtrn // this is not ideal method (but uncommonly used)
+let boolParse (path : string,r:Range) =
+    match path.Substring(r.Start, r.Finish - r.Start) with
+    | "true"  | "True"  | "TRUE"  -> true  |> rtrn
+    | "false" | "False" | "FALSE" -> false |> rtrn
```
[the remark this answers is not in the document]
Comment (gerardtoconnor) [reply frameworks/FSharp/Zebra/src/App/Parsers.fs]: The whole framework is in the PR as it is a new POC framework for F# that needs to perform well in benchmarking if it is going to be released as a full library. Everything in the library is handrolled for performance, the application is just Program.fs, the rest will be packaged into framework once validated. Is that ok?

[the remark this answers is not in the document]
Comment (michaelhixson) [reply frameworks/FSharp/Zebra/src/App/Parsers.fs]: Yeah, that's ok.

[the remark this answers is not in the document]
Comment (gerardtoconnor) [reply frameworks/FSharp/Zebra/src/App/Parsers.fs]: Thank you, provided it performs well, beats F# Giraffe, I can do PR when Framework is fully released.

**Pull request merged**